src/plugins/social-sharing-mixin.js

```diff
@@ -0,0 +1,136 @@
+import { getFullUrl } from '@/util/urlUtils';
+import clipboardCopy from 'clipboard-copy';
+
+/* This mixin depends on the component using it having the
+* following data or computed variables available:
+* this.shareMessage and this.shareLink
+* this.copyStatus is optional
+*/
+
+export default {
+	methods: {
+		handleFacebookResponse(trackingCategory) {
+			// Check for the route hash that facebook adds to the request
+			if (this.$route.hash === '#_=_') {
+				// Check for an error
+				const { error_code: code, error_message: message } = this.$route.query;
+				if (code) {
+					// The 4201 error code means the user pressed 'Cancel', so can be ignored
+					if (code !== '4201') {
+						this.$showTipMsg(`There was a problem sharing to Facebook: ${message}`, 'warning');
+					}
+					this.$kvTrackEvent(trackingCategory, 'fail', 'share-facebook');
+				} else {
+					this.$showTipMsg('Thanks for sharing to Facebook!');
+				}
+			}
+		},
+		/** displays the share pop up window for whatever service we are sharing on.  */
+		showSharePopUp(destinationUrl, thanksText) {
+			// This code taken from the twitter example in the docs
+			const width = 600;
+			const height = 420;
+			const winHeight = window.innerHeight;
+			const winWidth = window.innerWidth;
+			const left = Math.round((winWidth / 2) - (width / 2));
+			let top = 0;
+
+			if (winHeight > height) {
+				top = Math.round((winHeight / 2) - (height / 2));
+			}
+			window.open(
+				destinationUrl,
+				'intent',
+				// eslint-disable-next-line max-len
+				`scrollbars=yes,resizable=yes,toolbar=no,location=yes,width=${width},height=${height},left=${left},top=${top}`
+			);
+			this.$showTipMsg(thanksText);
+		},
+		twitterShareUrl({ utmCampaign, utmContent }) {
```

Comment:
The 3 functions work the same way. Take the url to be shared, first, add utm params. Then run that url as the input param for adding the params required by the social media platform.